lexer/lexer_test.go

```diff
@@ -256,3 +256,72 @@ func TestNextTokenCommentEOLMode(t *testing.T) {
 		}
 	}
 }
+
+func TestReadFloatNumber(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{".", "."},  // not a valid float number yet we consume and it'll fail to convert later.
+		{".a", "."}, // not a valid float number yet we consume and it'll fail to convert later.
+		{"123.", "123."},
+		{".5", ".5"},
+		{"100.56", "100.56"},
+		{"1e3", "1e3"},
```

Comment:
Please add 1e1_000 to test 1e1000

Reply:
max exponent on float64 is 308 yet sure I can put more, the lexer won't mind (1e1_000 is no different than 1e1_2_3)

Reply:
done nevertheless